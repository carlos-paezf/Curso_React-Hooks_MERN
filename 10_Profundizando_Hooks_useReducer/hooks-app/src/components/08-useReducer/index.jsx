export { TodoApp } from "./TodoApp";
export { TodoList } from "./TodoList";
export { TodoListItem } from "./TodoListItem";
export { TodoAdd } from "./TodoAdd";