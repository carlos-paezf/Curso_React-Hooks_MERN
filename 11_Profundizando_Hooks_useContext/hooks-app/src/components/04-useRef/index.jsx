export { FocusScreen } from './FocusScreen'
export { RealExampleRef } from './RealExampleRef'