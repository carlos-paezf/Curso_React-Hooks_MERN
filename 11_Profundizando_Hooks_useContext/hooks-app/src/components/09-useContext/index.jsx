export { AppRouter } from "./AppRouter";

export { AboutScreen } from "./AboutScreen";
export { HomeScreen } from "./HomeScreen";
export { LoginScreen } from "./LoginScreen";
export { MainApp } from "./MainApp";

export { NavBar } from "./NavBar";