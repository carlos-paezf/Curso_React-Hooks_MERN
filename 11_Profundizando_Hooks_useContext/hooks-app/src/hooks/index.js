/* The useFetch hook is a function that takes in a URL and returns a response. 
The useForm hook is a function that takes in a form and returns a form. 
The useCounter hook is a function that takes in a number and returns a number. */
export { useFetch } from "./useFetch";
export { useForm } from "./useForm";
export { useCounter } from "./useCounter";