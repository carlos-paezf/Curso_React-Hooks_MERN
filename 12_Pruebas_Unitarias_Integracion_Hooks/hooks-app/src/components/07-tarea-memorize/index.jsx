export { Hijo } from "./Hijo";
export { Padre } from "./Padre";