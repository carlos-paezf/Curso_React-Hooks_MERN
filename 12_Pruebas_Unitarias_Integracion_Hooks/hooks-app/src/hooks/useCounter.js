import { useState } from "react";

/**
 * @param [initialState=10] - The initial state of the counter.
 * @returns The state, increment, decrement, and reset functions.
 */
export const useCounter = (initialState = 10) => {
    const [state, setState] = useState(initialState);

    const increment = (factor = 1) => setState(state + factor)

    const decrement = (factor = 1) => setState(state - factor)

    const reset = () => setState(initialState)

    return { state, increment, decrement, reset };
};
