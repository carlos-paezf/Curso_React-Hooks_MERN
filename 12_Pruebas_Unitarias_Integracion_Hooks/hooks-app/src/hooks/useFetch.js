import { useEffect, useRef, useState } from "react";

/**
 * @param url - the URL to fetch
 * @returns The state object.
 */
export const useFetch = (url) => {

    const isMounted = useRef(true);

    const [state, setState] = useState({
        data: null,
        loading: true,
        error: null
    });

    useEffect(() => {
        return () => {
            isMounted.current = false
        }
    }, []);


    useEffect(() => {
        setState({ data: null, loading: true, error: null })
        fetch(url)
            .then(res => res.json())
            .then(data => {
                if (isMounted.current) {
                    setState({
                        data,
                        loading: false,
                        error: null
                    })
                } else {
                    console.log('setState no se llamo')
                }
            })
            .catch(error => {
                setState({
                    data: null,
                    loading: false,
                    error: 'No se pudo cargar la data'
                })
            })
    }, [url]);

    return state
}
