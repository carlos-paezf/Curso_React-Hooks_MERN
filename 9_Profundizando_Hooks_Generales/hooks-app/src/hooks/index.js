export { useFetch } from "./useFetch";
export { useForm } from "./useForm";
export { useCounter } from "./useCounter";