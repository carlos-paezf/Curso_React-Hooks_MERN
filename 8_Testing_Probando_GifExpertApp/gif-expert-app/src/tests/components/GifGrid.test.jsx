import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import { GifGrid } from '../../components';
import { useFetchGifs } from '../../hooks';

jest.mock('../../hooks/useFetchGifs')


describe('Pruebas al componente <GifGrid />', () => {
    const category = 'test'
    
    test('Debe mostrarse correctamente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        }) 
        const wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper).toMatchSnapshot()
    });

    test('Debe de mostrar items cuando se cargan imágenes useFetchGifs', () => {
        const gifs = [{
            id: 'ABX',
            url: 'url aquí',
            title: 'título del gif'
        }]
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })
        const wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper).toMatchSnapshot()

        expect(wrapper.find('p').exists()).toBe(false)
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
    })
});
