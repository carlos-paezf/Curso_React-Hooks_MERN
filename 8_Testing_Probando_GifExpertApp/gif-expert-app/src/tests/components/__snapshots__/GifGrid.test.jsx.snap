// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente <GifGrid /> Debe de mostrar items cuando se cargan imágenes useFetchGifs 1`] = `
<Fragment>
  <h3>
    test
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABX"
      key="ABX"
      title="título del gif"
      url="url aquí"
    />
  </div>
</Fragment>
`;

exports[`Pruebas al componente <GifGrid /> Debe mostrarse correctamente 1`] = `
<Fragment>
  <h3>
    test
  </h3>
  <p
    className="card animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
